examples/demo/updater/updater_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updater
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"sync"
+	"testing"
+)
+
+func TestUpdater(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	latest := make(chan string)
+
+	base := New(ctx, func(b []byte) {
+		latest <- string(b)
+	})
+
+	child := NewNested(ctx, base.ForField("Foo"))
+
+	l := <-latest
```

Comment:
If something was wrong and the value was not being set - wouldnt this block instead of failing?

Reply:
The test would deadlock and therefor fail.

Reply:
Would you rather reasonably timeout and fail? than deadlock and fail? I think deadlock and fail will have no good context on why it failed - and am not sure of how long before the framework decides it failed - for our CIs etc.

Reply:
Deadlocks have nice stack traces: https://play.golang.org/p/u7hcDtANkNl